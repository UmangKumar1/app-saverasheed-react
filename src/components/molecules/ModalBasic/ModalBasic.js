import { ClickAwayListener } from '@material-ui/core'
import React from 'react'
import styled, { keyframes } from 'react-emotion'
import PropTypes from 'prop-types'

import { noop } from '../../../util/ramda-extra'
import theme from '../../../theme'

const { any, func, string } = PropTypes

const {
  palette: {
    primary: { main: primaryMain, dark: primaryDark },
  },
} = theme

const ModalBasicRoot = styled('div')({
  position: 'fixed',
  top: 0,
  right: 0,
  bottom: 0,
  left: 0,
  zIndex: 2,
})

const screenAnim = keyframes`
0% {
  background-position: 0 0;
}

100% {
  background-position: 0 30px;
}
`

const PopupWrapper = styled('div')({
  background: `linear-gradient(to bottom, ${primaryDark}, ${primaryDark} 90%, ${primaryMain} 90%, ${primaryMain})`,
  backgroundSize: '100% 30px',
  animation: `${screenAnim} 1.5s linear infinite`,
  padding: 8,
  pointerEvents: 'all',
  overflow: 'scroll',
  borderRadius: 10,
  position: 'absolute',
  left: 5,
  right: 5,
})

const Popup = styled('div')({
  background: primaryMain,
  zIndex: 1,
  padding: '10px 20px',
  borderRadius: 5,
})

const ModalBasic = ({
  onClickAway = noop,
  children,
  className,
  popupWrapperClassName,
  popupClassName,
  onClick,
}) => (
  <ModalBasicRoot className={className}>
    <ClickAwayListener onClickAway={onClickAway}>
      <PopupWrapper onClick={onClick} className={popupWrapperClassName}>
        <Popup className={popupClassName}>{children}</Popup>
      </PopupWrapper>
    </ClickAwayListener>
  </ModalBasicRoot>
)

ModalBasic.propTypes = {
  onClickAway: func,
  children: any.isRequired,
  className: string,
  popupWrapperClassName: string,
  popupClassName: string,
  onClick: func,
}

export default ModalBasic
