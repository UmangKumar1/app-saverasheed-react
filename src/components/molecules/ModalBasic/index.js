export { default, default as ModalBasic } from './ModalBasic'
