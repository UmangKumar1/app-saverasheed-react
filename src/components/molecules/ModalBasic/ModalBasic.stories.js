import React from 'react'
import { storiesOf } from '@storybook/react'
import ModalBasic from '.'

storiesOf('ModalBasic', module).add('off', () => (
  <ModalBasic>Children</ModalBasic>
))
