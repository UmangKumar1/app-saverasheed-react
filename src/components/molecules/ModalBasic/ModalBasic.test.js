import React from 'react'
import { shallow } from 'enzyme'
import ModalBasic from '.'

const wrap = (props = {}) => shallow(<ModalBasic {...props} />)

describe('ModalBasic', () => {
  it('renders children when passed in', () => {
    const wrapper = wrap({
      children: 'children',
    })
    expect(wrapper).toMatchSnapshot()
  })
})
