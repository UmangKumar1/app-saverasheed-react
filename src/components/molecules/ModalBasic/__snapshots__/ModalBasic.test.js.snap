// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBasic renders children when passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalBasic>
    children
  </ModalBasic>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ClickAwayListener
        mouseEvent="onMouseUp"
        onClickAway={[Function]}
        touchEvent="onTouchEnd"
      >
        <Styled(div)>
          <Styled(div)>
            children
          </Styled(div)>
        </Styled(div)>
      </ClickAwayListener>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(div)>
          <Styled(div)>
            children
          </Styled(div)>
        </Styled(div)>,
        "mouseEvent": "onMouseUp",
        "onClickAway": [Function],
        "touchEvent": "onTouchEnd",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(div)>
            children
          </Styled(div)>,
          "className": undefined,
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "children",
            "className": undefined,
          },
          "ref": null,
          "rendered": "children",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ClickAwayListener
          mouseEvent="onMouseUp"
          onClickAway={[Function]}
          touchEvent="onTouchEnd"
        >
          <Styled(div)>
            <Styled(div)>
              children
            </Styled(div)>
          </Styled(div)>
        </ClickAwayListener>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(div)>
            <Styled(div)>
              children
            </Styled(div)>
          </Styled(div)>,
          "mouseEvent": "onMouseUp",
          "onClickAway": [Function],
          "touchEvent": "onTouchEnd",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(div)>
              children
            </Styled(div)>,
            "className": undefined,
            "onClick": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "children",
              "className": undefined,
            },
            "ref": null,
            "rendered": "children",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
